api/chaos/serverexperimentation/v1/serverexperimentation.proto

```diff
@@ -56,3 +56,128 @@ enum FaultInjectionCluster {
   // Faults are injected to ingesss traffic by an upstream service(s).
   FAULTINJECTIONCLUSTER_UPSTREAM = 2;
 }
+
+// The configuration of an HTTP fault.
+message HTTPFaultConfig {
+  oneof fault {
+    option (validate.required) = true;
+
+    // The abort fault.
+    AbortFault abort_fault = 1;
+    // The latency fault.
+    LatencyFault latency_fault = 2;
+  }
+}
+
+// The definition of an abort fault.
+message AbortFault {
+  // The targeting of the fault describing what requests faults should be applied to.
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  // The percentage of requests the fault should be applied to.
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  // The abort status to apply.
+  FaultAbortStatus fault_abort_status = 3 [ (validate.rules).message.required = true ];
+}
+
+// The definition of a latency fault.
+message LatencyFault {
+  // The targeting of the fault describing where the faults should be applied.
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  // The percentage of requests the fault should be applied to.
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  // The latency duration to apply.
+  FaultLatencyDuration fault_latency_duration = 3 [ (validate.rules).message.required = true ];
+}
+
+// The fault targeting that describes a part of the system that's responsible for enforcing a fault.
+message FaultTargeting {
+  // A fault enforcer.
+  oneof enforcer {
+    option (validate.required) = true;
+
+    // Enforce a fault on an upstream.
+    UpstreamEnforcing upstream = 1;
+    // Enfore a fault on a downstream.
+    DownstreamEnforcing downstream = 2;
+  }
+}
+
+/// Enforce faults on an upstream.
+message UpstreamEnforcing {
+  // An upstream responsible for enforcing a fault.
+  oneof upstream_type {
+    option (validate.required) = true;
+
+    // A single upstream cluster.
+    SingleCluster upstream_cluster = 1;
+    // A partial single upstream cluster.
+    PartialSingleCluster upstream_partial_single_cluster = 2;
+  }
+
+  // A downstream responsible for enforcing a fault.
+  oneof downstream_type {
+    option (validate.required) = true;
+
+    // A single downstream cluster.
+    SingleCluster downstream_cluster = 3;
+  }
+}
+
+/// Enforce faults on a downstream.
+message DownstreamEnforcing {
+  // An upstream responsible for enforcing a fault.
+  oneof upstream_type {
+    option (validate.required) = true;
+
+    // A single upstream cluster.
+    SingleCluster upstream_cluster = 1;
+  }
+
+  // A downstream responsible for enforcing a fault.
+  oneof downstream_type {
+    option (validate.required) = true;
+
+    // A single downstream cluster.
+    SingleCluster downstream_cluster = 2;
+  }
+}
+
+// A single cluster.
```

Comment:
Maybe expand this to something like `A single cluster that is partaking in the fault injection`